astropy/coordinates/sky_coordinate.py

```diff
@@ -483,6 +485,60 @@ def transform_to(self, frame, merge_attributes=True):
             frame_kwargs.pop(attr)
         return self.__class__(new_coord, **frame_kwargs)
 
+    @u.quantity_input(dt=(u.year, None))
+    def move_to(self, new_obstime=None, dt=None):
+        """
+        Update the position of the source represented by this coordinate object
+        to a new specified time using the velocity and assuming linear motion.
+
+        This is sometimes referred to as an "epoch transformation."
+
+        Parameters
+        ----------
+        new_obstime : `~astropy.time.Time`, optional
+            The time at which to evolve the position to.
+        dt : `~astropy.units.Quantity`, optional
+            A time to evolve as a `Quantity` object.
+        """
+
+        if (new_obstime is None and dt is None or
+                new_obstime is not None and dt is not None):
+            raise ValueError("You must specify one of `new_obstime` or `dt`, "
+                             "but not both.")
+
+        # Validate that we have velocity info
+        if 's' not in self.frame.data.differentials:
+            raise ValueError('SkyCoord requires velocity data to evolve the '
+                             'position.')
+
+        if 'obstime' in self.frame.frame_attributes:
+            raise NotImplementedError("Updating the coordinates in a frame "
+                                      "with explicit time dependence is "
+                                      "currently not supported. If you would "
+                                      "like this functionality, please open an "
+                                      "issue on github:\n"
+                                      "https://github.com/astropy/astropy")
+
+        if dt is None:
+            # If no obstime already on this object, raise error: we need to know the
+            # time / epoch at which the the position/velocity were measured
+            dt = (new_obstime - self.obstime)
+
+        cart = self.frame.represent_as('cartesian')
+        pos = cart.without_differentials()
+        vel = cart.differentials['s']
+
+        new_rep = pos + vel.to_cartesian() * dt
```

Comment:
I worry if we should be 'doing this right' from the start, by which I mean accounting for relativity and light travel time? ```erfa``` has a ```starpm``` routine to handle this. We could either:

1. transform coord to ICRS and use ```starpm```
1. implement our own transformation and test against ```starpm```

For the size of the impact of not 'doing it right', see the comment on the tests...